content-src/selectors/siteMetadataSelectors.js

```diff
@@ -8,7 +8,12 @@ function selectSiteProperties(site) {
   const metadataFavicon = site.favicons && site.favicons[0] && site.favicons[0].url;
   const favicon = site.favicon_url || metadataFavicon || site.favicon;
   const parsedUrl = site.parsedUrl || urlParse(site.url || "");
-  const label = prettyUrl(parsedUrl.hostname);
+
+  // Remove the eTLD (e.g., com, net) and the preceding period from the hostname
+  const eTLDLength = (site.eTLD || "").length;
+  const eTLDExtra = eTLDLength > 0 ? -(eTLDLength + 1) : Infinity;
+  const label = prettyUrl(parsedUrl.hostname).slice(0, eTLDExtra);
```

Comment:
Will `parsedUrl.hostname` work here? It seems like we _may_ possibly fall back to "" above on L10 above.